lib/semantics/resolve-names.cc

```diff
@@ -3568,6 +3571,78 @@ bool DeclarationVisitor::Pre(const parser::StructureConstructor &x) {
   return false;
 }
 
+bool DeclarationVisitor::Pre(const parser::BasedPointerStmt &x) {
+  for (const parser::BasedPointer &bp : x.v) {
+    const parser::ObjectName &pointerName{std::get<0>(bp.t)};
+    const parser::ObjectName &pointeeName{std::get<1>(bp.t)};
+    auto *pointer{FindSymbol(pointerName)};
+    if (!pointer) {
+      pointer = &MakeSymbol(pointerName, ObjectEntityDetails{});
+    } else if (pointer->has<UseDetails>()) {
+      Say(pointerName,
+          "'%s' cannot be a Cray pointer as it is use-associated"_err_en_US);
+    } else if (!ConvertToObjectEntity(*pointer) || IsParameter(*pointer)) {
```

Comment:
Just a heads up that I renamed `IsParameter` to `IsNamedConstant` so you'll have a logical conflict once you rebase/merge with master.